### PR TITLE
Fix #9772 - AirLoop Component Load Summary report for Cooling Peak Conditions always shows Mixed Air Temperature = 0

### LINKED ISSUE #9760: Incorrect spelling of "Oputput:Table:Annual" to be corrected to "Output:Table:Annual"
Hi,
There is a spelling error in the code that relates to a "Oputput:Table:Annual" in [src/EnergyPlus/OutputReportTabularAnnual.cc](https://github.com/NREL/EnergyPlus/blob/de5b76fcb67d332838150389af1e6a17c8d1eee8/src/EnergyPlus/OutputReportTabularAnnual.cc)  lines 131 and 132. 

This makes EnergyPlus return a fatal error even if there is an Output:Table:Annual with all the fields completed.

image

### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### LINKED ISSUE #9772: AirLoop Component Load Summary report for Cooling Peak Conditions always shows Mixed Air Temperature = 0
Issue overview
--------------
[Unmet Hours post](https://unmethours.com/question/85350/why-is-autosizing-leading-to-undersized-system/?comment=85432#post-id-85432) questions why the Mixed Air Temperature = 32 F (0 C) for the simulation. This reporting issue was replicated using example file 5ZoneAirCooled by adding that table report. 

Code review shows this output was never set/initialized (by searching for mixAirTemp).

    void OutputCompLoadSummary(,,,)
            if (kind == OutputType::AirLoop) {
                tableBody(1, 9) = RealToStr(curCompLoad.mixAirTemp, 2); // mixed air temperature - not for zone or facility
            }

image

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/9000s/9772/5ZoneAirCooled_mod.idf
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #9772
 - Fixes #9760

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 1]
```diff
@@ -7251,6 +7264,28 @@ TEST_F(SQLiteFixture, OutputReportTabular_WriteLoadComponentSummaryTables_AirLoo
...
+        {
+            const std::string queryHeat = R"sql(
+            SELECT Value From TabularDataWithStrings
+             WHERE TableName = 'Heating Peak Conditions'
+                AND ReportName = 'AirLoop Component Load Summary'
+                AND RowName = 'Mixed Air Temperature')sql";
+
+            EXPECT_EQ(heatMixTempSysIP, execAndReturnFirstDouble(queryHeat));
+        }
+    }
```
Comment (rraustad) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:7291]: I assume this is failing now? until you fix this.

Comment (jmarrec) [reply tst/EnergyPlus/unit/OutputReportTabular.unit.cc:7291]: Yes, it returns 32 (in F, 0 in C)

File: src/EnergyPlus/OutputReportTabular.cc [hunk 2]
```diff
@@ -15710,6 +15710,11 @@ void WriteLoadComponentSummaryTables(EnergyPlusData &state)
                         airLoopCoolTable.diffDesignPeak = airLoopHeatTable.designPeakLoad - airLoopHeatTable.peakDesSensLoad;
                     }
                 }
+
+                // Coincident and NonCoincident alike
+                airLoopCoolTable.mixAirTemp = finalSysSizing.MixTempAtCoolPeak;
+                airLoopHeatTable.mixAirTemp = finalSysSizing.HeatMixTemp;
+
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabular.cc:15717]: Get the entries. The ZoneSizingData struct does not store the airloop's mixed air temperature, so we just get the value for coincident and noncoincident in the same way, by retrieving it on the FinalSysSizing entry.

File: src/EnergyPlus/OutputReportTabularAnnual.cc [hunk 3]
```diff
@@ -129,7 +129,7 @@ void GetInputTabularAnnual(EnergyPlusData &state)
             for (jAlpha = 4; jAlpha <= numAlphas; jAlpha += 2) {
                 curVarMtr = alphArray(jAlpha);
                 if (curVarMtr.empty()) {
-                    ShowFatalError(state, "Blank report name in Oputput:Table:Annual");
+                    ShowFatalError(state, "Blank report name in Output:Table:Annual");
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabularAnnual.cc:132]: Typo from #9760

File: src/EnergyPlus/OutputReportTabular.cc (hunk 2 above)
Comment (rraustad) [reply src/EnergyPlus/OutputReportTabular.cc:15717]: This is set inside the loop 1 to n for air loops. Seems perfectly reasonable to me.

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 4]
```diff
@@ -7188,6 +7188,22 @@ TEST_F(SQLiteFixture, OutputReportTabular_WriteLoadComponentSummaryTables_AirLoo
     // same Design Days peak and timestep peak as the zone it serves. This is the critical part of the test
     state->dataSize->FinalSysSizing(1).loadSizingType = DataSizing::LoadSizing::Total;
 
+    // For #9772
+    state->dataSize->NumSysSizInput = 1;
```
Comment (rraustad) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:7192]: Now the loop where you gather the mixed temps will exectute and capture the correct data.

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc (hunk 1 above)
Comment (rraustad) [reply tst/EnergyPlus/unit/OutputReportTabular.unit.cc:7291]: And now a passing unit test proving this is fixed. I just have to run the unit tests and an example to verify.

**Comment by rraustad** [comment]
@jmarrec you reviewed the html reports? The unit tests do pass but I still see 0's in the cooling/heating AirLoop Component Load Summary?

image

**Comment by rraustad** [comment]
Or maybe not, I pulled the wrong branch, 9722.

**Comment by rraustad** [comment]
Considering the above as the develop branch (sure), this branch shows data for the Mixed Air Temperature for both Cooling and Heating Peak Conditions. Unit tests also run successfully on this branch.

image

**Comment by rraustad** [comment]
Given there are no diffs, I will suggest that we add these table reports to the example file 5ZoneAirCooled.

    Output:Table:SummaryReports,
      AllSummary,              !- Report 1 Name
      AirLoopComponentLoadSummary, <--- this fix is for these tables
      ZoneComponentLoadSummary;!- Report 2 Name

**Review by rraustad — APPROVED** [review]
Everything looks good. Clean fix to get these Mixed Air Temperature's to report.

**Comment by jmarrec** [comment]
> Or maybe not, I pulled the wrong branch, 9722.

I got scared I messed up for a minute :) Thanks for the review

**Comment by jmarrec** [comment]
> Given there are no diffs, I will suggest that we add these table reports to the example file 5ZoneAirCooled.
> 
> ```
> Output:Table:SummaryReports,
>   AllSummary,              !- Report 1 Name
>   AirLoopComponentLoadSummary, <--- this fix is for these tables
>   ZoneComponentLoadSummary;!- Report 2 Name
> ```

We could replace AllSummary with AllSummaryAndSizingPeriod or even AllSummaryMonthlyAndSizingPeriod
https://github.com/NREL/EnergyPlus/blob/4e7b9df50edb2c7bf9f3a38b29b747e1bef20d82/src/EnergyPlus/OutputReportTabular.cc#L1611-L1736

There are zero files that produce this report indeed.

These all come up empty

```shell
cd testfiles
grep -R AirLoopComponentLoadSummary
grep -R AllSummaryAndSizingPeriod
grep -R AllSummaryAndMonthlySizingPeriod
```

**Comment by rraustad** [comment]
@mjwitte would you agree with modifying 5ZoneAirCooled to report these other tables?

**Comment by mjwitte** [comment]
> @mjwitte would you agree with modifying 5ZoneAirCooled to report these other tables?

Yes, let's go with AllSummaryAndSizingPeriod. There's already one tetsfile with "AllMonthly" and several with AllSummaryAndMonthly.

**Comment by jmarrec** [comment]
Done in 93f2a83ac53914907d3435f3c2648e89408d8cae

File: testfiles/5ZoneAirCooled.idf [hunk 5]
```diff
@@ -3492,8 +3492,7 @@
     HTML;                    !- Column Separator
 
   Output:Table:SummaryReports,
-    AllSummary,              !- Report 1 Name
-    ZoneComponentLoadSummary;!- Report 2 Name
+    AllSummaryAndSizingPeriod; !- Report 1 Name
```
Comment (jmarrec) [inline testfiles/5ZoneAirCooled.idf:3495]: As requested @rraustad

**Comment by rraustad** [comment]
@Myoldmopar @mjwitte this is ready to merge once CI finishes, not that these results will show anything new.

**Comment by rraustad** [comment]
CI came back clean again, merging.

**Comment by rraustad** [comment]
I got a doc build workflow failure on the merge. I will disregard since no docs were changed here.

**Pull request merged**